### ISSUE TITLE
Button label not rendering without setting the long_name value

### ISSUE THREAD

**Issue by kostoskistefan** [Bodmer/TFT_eSPI#3274]
I've been testing out the TFT_eSPI library version 2.5.43 and encountered a strange problem. I am using PlatformIO with STM32F103C8T6 Blue Pill and an ILI9341 TFT Touchscreen connected via SPI.

I created a TFT_eSPI_Button, initialized it with the following label "<" and then when I ran the code, the button didn't have a label. 

Here is my initialization code:
```cpp
TFT_eSPI tft;
TFT_eSPI_Button button_navigate_left;

void ui_initialize(void)
{
    tft.init();

    uint16_t calibration_data[5] = { 225, 3442, 302, 3497, 1 };
    tft.setTouch(calibration_data);

    tft.setRotation(1);
    tft.setTextColor(TFT_WHITE, TFT_BLACK);
    tft.setTextSize(2);
    tft.setTextDatum(MC_DATUM);

    pinMode(TFT_LED_PIN, OUTPUT);
    digitalWrite(TFT_LED_PIN, LOW);

    button_navigate_left.initButton(&tft, 20, 20, 30, 30, TFT_WHITE, TFT_WHITE, TFT_BLACK, (char *) "<", 2);

    // ... other code
}
```

Here is where I render the buttons:

```cpp
void ui_render(void)
{
    button_navigate_left.drawButton(true);

    // ... other code
}
```

I noticed the long_name parameter in the drawButton function, which is an empty string by default. I tried passing an empty string to the function for completeness sake, but that didn't have any effect as expected. If I pass a non-empty string, everything works great. Then I went digging in the code of the library and realized that there might be a problem with the long_name comparison, here:

https://github.com/Bodmer/TFT_eSPI/blob/fae22f785f8b4f7970195487826a348d08d082ae/Extensions/Button.cpp#L85-L88

So, I commented out the if and else statements and only uncommented the following line and it worked:
```cpp
_gfx->drawString(_label, _x1 + (_w/2) + _xd, _y1 + (_h/2) - 4 + _yd);
 ```

Therefore, I came to the conclusion that there must be a problem with the comparison in the if statement. 
Long story short, changing the if statement to the following code fixes the problem:

```cpp
if (long_name.compareTo("") == 0)
    _gfx->drawString(_label, _x1 + (_w/2) + _xd, _y1 + (_h/2) - 4 + _yd);
else
    _gfx->drawString(long_name, _x1 + (_w/2) + _xd, _y1 + (_h/2) - 4 + _yd);
```

Something that is a bit strange to me is that it still doesn't work if I use the long_name.equals function. It only works with the compareTo function (and strcmp, but since it is an Arduino String type anyways, there is no point in using that).

I am happy to open a pull request if you want to, but first I wanted to hear what you think about this and whether it is something that I did wrong, or there is an actual bug in the library.
